ckanext/datarequests/auth.py

```diff
@@ -39,7 +40,25 @@ def _is_any_group_member(context):
     return user_name and authz.has_user_permission_for_some_org(user_name, 'read')
 
 
-@auth_allow_anonymous_access
+def auth_allow_superadmin_only(action):
+    ''' Flag an auth function as requiring a superadmin user
+
+    This means that check_access will automatically raise a NotAuthorized
+    exception if a superadmin user is not provided in the context.
+    '''
+    def decorated(context, data_dict=None):
+        user_name = context.get('user')
+        try:
+            user = get_action('user_show')(context, {'id': user_name})
+            if not user.get('sysadmin', False):
+                raise NotAuthorized('You must be a superadmin to perform this action.')
+        except NotFound:
+            raise NotAuthorized('User not found.')
+        return action(context, data_dict)
+    return decorated
```

Comment:
Don't know much about CKAN code but why aren't we using helper methods like [`is_sysadmin`](https://github.com/ckan/ckan/blob/442f03004ddea16b9625281cf29a568b21c7196b/ckan/authz.py#L144) for this?

Reply:
Nice catch, just seen most likely can achieve all the above on the templates without a new decorator